ESP Thread Border Router, board and shield support

marekmatej opened the pull request:
Adding support for board and shield:
- ESP Thread Border Router / ZigBee Gateway
- ESP Thread Border Router ethernet shield


https://builds.zephyrproject.io/zephyr/pr/99704/docs/boards/espressif/esp_thread_br/doc/index.html

https://builds.zephyrproject.io/zephyr/pr/99704/docs/boards/shields/esp_thread_br_ethernet/doc/index.html

marekmatej committed:
boards,shields: ESP Thread Border Router with ethernet

Add support for ESP Thread Border Router.
Add Ethernet shield for the ESP Thread Border Router.

Signed-off-by: Marek Matej <marek.matej@espressif.com>

sylvioalves commented:
Fix `appcpu.yaml` file name: 
`esp_thread_br_zigbeegw_appcpu.yaml` -> `esp_threadbr_zigbeegw_appcpu.yaml`

sylvioalves commented on boards/espressif/esp_threadbr_zigbeegw/esp_threadbr_zigbeegw_appcpu.dts:
```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2024 Espressif Systems (Shanghai) Co., Ltd.
```
update copyright to `2026` in all new files.

marekmatej committed:
tests: boards: espresif: add esp_threadbr

Add new test case to cover esp_threadbr with ethernet shield.

Signed-off-by: Marek Matej <marek.matej@espressif.com>

sylvioalves commented on boards/espressif/esp_thread_br/esp_thread_br-pinctrl.dtsi:
```diff
@@ -0,0 +1,71 @@
+		};
+	};
+
+	spim2_default: spim2_default {
```
I don't get this. `esp_thread_br` pins defines SPI2 to use `GPIO12/13/11/10` but then the shield overlay redefines the whole mapping to `GPIO21/38/45` + cs-gpio to 41. Is this the expected? Sounds one or the other is not needed or wrong?

sylvioalves commented on boards/espressif/esp_thread_br/esp_thread_br_appcpu.dts:
```diff
@@ -0,0 +1,31 @@
+ */
+/dts-v1/;
+
+#include <espressif/esp32s3/esp32s3_wroom_n8.dtsi>
```
If all the esp_thread_br board are N8R2, why not enabling spiram by default?

marekmatej commented on boards/espressif/esp_thread_br/doc/index.rst:
```diff
@@ -0,0 +1,55 @@
+
+* The host Wi-Fi SoC, ESP32-S3.
+* The radio co-processor (RCP), which is an ESP32-H2 series SoC.
+  The RCP enables the Border Router to access the 802.15.4 physical and MAC layer.
```
I intentionally did not put pin mappings, as this is covered in the provided documentation.

Usage is again documented in source documentation, and I would rather not copy and paste.

maass-hamburg commented on boards/espressif/esp_thread_br/esp_thread_br_procpu.yaml:
```diff
@@ -0,0 +1,20 @@
+toolchain:
+  - zephyr
+supported:
+  - gpio
```
add `- netif:wifi` ?

marekmatej committed:
soc: espressif: fix llext_heap missing

Some test failed because of orphaned section caused by missing
sections. This fixes the issue by including common-noinit.ld.

Signed-off-by: Marek Matej <marek.matej@espressif.com>

sylvioalves commented on soc/espressif/esp32/default.ld:
```diff
@@ -787,6 +787,8 @@ SECTIONS
     _bss_end = ABSOLUTE(.);
   } GROUP_LINK_IN(RAMABLE_REGION)
 
+  #include <zephyr/linker/common-noinit.ld>
```
I think all the ` #include <zephyr/linker/common-noinit.ld>` isn't correct. It is redefining another `NOINIT` area and also replacing existing `*(.noinit)` in the linker. It means ESP32 and ESP32-S3 `#ifdef CONFIG_ESP32_WIFI_NET_ALLOC_SPIRAM` won't met any longer.

The proper way to fix the issue is adding `#include <snippets-noinit.ld>` right after `*(.noinit.*)`, except for ESP32 and ESP32-S3 that it should be placed before `__dram_noinit_end = ABSOLUTE(.);`.
Please re-check.

sylvioalves commented on boards/espressif/esp_threadbr/esp_threadbr_connector.dtsi:
```diff
@@ -0,0 +1,31 @@
+			   /* GPIO37 */ <17 0 &gpio1 5 0>, <18 0 &gpio1 6 0>,   /* GPIO38 */
+			   /* GPIO39 */ <19 0 &gpio1 7 0>, <20 0 &gpio1 8 0>,   /* GPIO40 */
+			   /* GPIO41 */ <21 0 &gpio1 9 0>, <22 0 &gpio1 10 0>,  /* GPIO42 */
+			   /* GPIO43 */ <23 0 &gpio1 11 0>, <24 0 &gpio1 12 0>, /* GPIO44 */
```
I see this is linked to the whole board header connector. Well, I would say that this is user-dependent choice after all. There must be some awareness about pinout. Please dismiss this.
